File: src/test/java/at/fhhagenberg/sqe/api/ServicedFloorServiceTest.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.di.DI;
import at.fhhagenberg.sqe.di.RealIElevator;
import at.fhhagenberg.sqe.entity.ServicedFloor;
import com.google.inject.Injector;
import com.google.inject.Key;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import sqelevator.IElevator;
import java.rmi.RemoteException;
import java.util.List;
import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.verify;

public class ServicedFloorServiceTest {

    private ServicedFloorService service;
    private IElevator realIElevator;

    @BeforeEach
    public void setUp() {
        Injector injector = DI.getInjector();
        service = injector.getInstance(ServicedFloorService.class);
        realIElevator = injector.getInstance(Key.get(IElevator.class, RealIElevator.class));
    }

    @Test
    public void testGetAll() throws RemoteException {
        List<ServicedFloor> servicedFloors = service.getAll(0);

        assertEquals(2, servicedFloors.size());
    }

    @Test
    public void testGet() throws RemoteException {
        ServicedFloor servicedFloor = service.get(0, 0);

        assertEquals(0, servicedFloor.getElevatorNumber());
        assertEquals(true, servicedFloor.isServiced());
        assertEquals(0, servicedFloor.getElevatorNumber());
    }

    @Test
    public void testUpdateServicedFloor() throws RemoteException {
        ServicedFloor servicedFloor = service.get(0, 0);
        servicedFloor.setServiced(false);
        service.updateServicedFloor(servicedFloor);

        verify(realIElevator).setServicesFloors(0, 0, false);
    }
}

File: src/test/kotlin/at/fhhagenberg/sqe/api/ServicedFloorServiceTest.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.di.TestDI
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.mockito.Mockito
import sqelevator.ConnectableIElevator
import sqelevator.IElevator
import java.rmi.RemoteException
import kotlin.test.*

class ServicedFloorServiceTest {

    private lateinit var service: ServicedFloorService
    private lateinit var realIElevator: IElevator

    @BeforeEach
    fun setUp() {
        val injector = TestDI.createMockInjector()
        service = injector.getInstance(ServicedFloorService::class.java)
        realIElevator = injector.getInstance(ConnectableIElevator::class.java)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetAll() {
        val servicedFloors = service.getAll(0)

        assertEquals(2, servicedFloors.size)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetAllNegativeElevators() {
        Mockito.`when`(realIElevator.elevatorNum).thenReturn(-1)

        assertEquals(0, service.getAll(0).size)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetAllNegativeFloors() {
        Mockito.`when`(realIElevator.floorNum).thenReturn(-1)

        assertEquals(0, service.getAll(0).size)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGet() {
        val servicedFloor = service.get(0, 0)!!

        assertEquals(0, servicedFloor.elevatorNumber)
        assertEquals(true, servicedFloor.isServiced)
        assertEquals(0, servicedFloor.elevatorNumber)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetInvalid() {
        assertNull(service.get(-1, 0))
        assertNull(service.get(0, -1))
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetNegativeElevators() {
        Mockito.`when`(realIElevator.elevatorNum).thenReturn(-1)

        assertNull(service.get(0, 0))
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetNegativeFloors() {
        Mockito.`when`(realIElevator.floorNum).thenReturn(-1)

        assertNull(service.get(0, 0))
    }

    @Test
    @Throws(RemoteException::class)
    fun testUpdateServicedFloor() {
        val servicedFloor = service.get(0, 0)!!
        service.updateServicedFloor(servicedFloor, false)

        Mockito.verify(realIElevator).setServicesFloors(0, 0, false)
    }

    @Test
    @Throws(RemoteException::class)
    fun testUpdateServicedFloorError() {
        Mockito.`when`(realIElevator.setServicesFloors(0, 0, false)).thenThrow(RemoteException())
        val servicedFloor = service.get(0, 0)!!

        assertThrows<RemoteException> {
            service.updateServicedFloor(servicedFloor, false)
        }
    }
}